apps/www/components/Discussion/overlays/DiscussionPreferences.js

```diff
@@ -356,6 +357,13 @@ const DiscussionPreferencesEditor = ({
                   setState((curr) => ({
                     ...curr,
                     anonymity: val,
+                    credential:
+                      val === false // if not anonymous keep or set saved credential
+                        ? state.credential ||
+                          userPreference?.credential?.description ||
+                          autoCredential?.description ||
+                          ''
+                        : '',
```

Comment:
We should limit the logic here to something really simple that focuses on removing auto credentials when going anonymous, something like this:
```suggestion
                    credential:
                      val && curr.credential === autoCredential?.description 
                        ? '' // remove auto credential when setting anonymous to true
                        :  curr.credential
```
(untested)

Anything more complex probably causes more problems than it solves. It's just not an expected behaviour to mutate other fields when unchecking a box and users might not notice what is being set further above (e.g. on mobile).

Reply:
I adapted your change a bit to unset the credential if it's equal to the initialStates credential.